### PR TITLE
SP-59-offender-summary-sentence-plan-section

### PR DESCRIPTION
**Pull request by jonhindle**

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -193,17 +192,22 @@ module.exports = function createApp({ signInService, formService }) {
     res.redirect(authLogoutUrl)
   })
   app.use(authenticationMiddleware())
-  app.get('/', (req, res) => res.render('formPages/crn_search'))
-  app.get('/offender_summary/:crn(x\\d{6})', (req, res) => {
+  app.get('/', (req, res) => res.render('formPages/crnSearch'))
+  app.get('/offenderSummary/crn/:crn(x\\d{6})', (req, res) => {
```
Comment (petergphillips) [inline server/app.js]: I don't know the context and nothing to do with me, but for our pages urls should normally be separated by hyphens and all lower case - https://www.gov.uk/guidance/content-design/url-standards-for-gov-uk

Comment (jonhindle) [reply server/app.js]: We had that debate yesterday, didn't spot the official guidance, just looked at the other projects. I'll update to follow the official guidance 👍

File: server/views/pages/offenderSummary.html [hunk 2]
```diff
@@ -69,12 +69,27 @@ <h2 class="govuk-heading-m">Offence and sentence</h2>
...
       firstCellIsHeader: true,
       rows: [
-        [{text: "Date plan started"}, {text: "place holder"}]
+        [{text: "Date plan started"}, {text: sentencePlan.createdDate}]
       ]
     }) }}
+    
+    {% endif %}
+    {{ govukButton({
+      text: "Create Sentence Plan",
```
Comment (petergphillips) [inline server/views/pages/offenderSummary.html]: normally text only starts with a capital letter and the other words are all lower case - see https://design-system.service.gov.uk/components/button/

File: server/views/pages/offenderSummary.html [hunk 3]
```diff
@@ -69,12 +69,27 @@ <h2 class="govuk-heading-m">Offence and sentence</h2>
   </section>
   <section class="govuk-grid-row">
     <h2 class="govuk-heading-m">Sentence Plan</h2>
```
Comment (petergphillips) [inline server/views/pages/offenderSummary.html]: should be `Sentence plan` to follow gov uk convention

File: server/services/offenderSummaryService.js [hunk 4]
```diff
@@ -1,10 +1,20 @@
...
+const { getOffenderData, getSentencePlan } = require('../data/oasysClient')
 const logger = require('../../log.js')
 
 const offenderSummaryData = async (crn, callback) => {
   logger.info(`Calling getOffenderData(${crn})`)
   try {
     const result = await getOffenderData(crn)
+
+    try {
+      if (result.identifiers) {
```
Comment (petergphillips) [inline server/services/offenderSummaryService.js]: don't need nested `if`

File: server/services/offenderSummaryService.js [hunk 5]
```diff
@@ -1,10 +1,20 @@
...
+        if (result.identifiers.nomisId) {
+          result.sentencePlan = await getSentencePlan('nomisId', result.identifiers.nomisId)
+        }
+      }
+    } catch (err) {
+      logger.info('No sentence plan returned.')
```
Comment (petergphillips) [inline server/services/offenderSummaryService.js]: not great fan of catching all and then not logging anything to say what has gone wrong

Comment (petergphillips) [reply server/services/offenderSummaryService.js]: should it be `error` too?

Comment (jonhindle) [reply server/services/offenderSummaryService.js]: No sentence-plan returned (404) is a valid scenario from the API.

Comment (jonhindle) [reply server/services/offenderSummaryService.js]: 🤔 Although logging out the error may have value 👍

File: server/views/formPages/crnSearch.html [hunk 6]
```diff
@@ -24,7 +24,7 @@
 <script>
   function goToCrnPage(event){
     var crn = document.getElementById("crnNumber").value
-    if(crn.match(/^\d{6}$/)) document.location = '/offender_summary/x'+crn
+    if(crn.match(/^\d{6}$/)) document.location = '/offenderSummary/crn/x'+crn
```
Comment (petergphillips) [inline server/views/formPages/crnSearch.html]: shouldn't this be `/offender-summary`?

File: server/views/formPages/sentencePlan.html [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+      href: "/offenderSummary/crn/" + crn
+    }
+  ]
+}) }}
+{% endblock %}
+
+{% block content %}
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-full">
+    <h1 class="govuk-heading-xl">Sentence Plan for {{crn}}</h1>
```
Comment (petergphillips) [inline server/views/formPages/sentencePlan.html]: `Sentence plan for...`

File: server/views/formPages/sentencePlan.html [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+{% block beforeContent %}
+{{ govukBreadcrumbs({
+  items: [
+    {
+      text: "Home",
+      href: "#"
+    },
+    {
+      text: "Offender summary",
+      href: "/offenderSummary/crn/" + crn
```
Comment (petergphillips) [inline server/views/formPages/sentencePlan.html]: offender-summary?

File: server/views/formPages/sentencePlan.html (hunk 7 above)
Comment (petergphillips) [inline server/views/formPages/sentencePlan.html]: Sentence plan for...

File: server/services/offenderSummaryService.js (hunk 5 above)
Comment (petergphillips) [reply server/services/offenderSummaryService.js]: Is there no way that we can distinguish between a 404 and an actual error calling the service then?

File: server/views/formPages/sentencePlan.html (hunk 8 above)
Comment (petergphillips) [reply server/views/formPages/sentencePlan.html]: is there no testing in this project :disappointed: ?  Would have thought that :point_up: would have caused a test to fail.  Should really have unit and integration tests...

**Pull request merged**